### PR TITLE
parser: implement Restore for ColumnDef and ColumnOption

### PR DESCRIPTION
**Pull request by feloxx**
1. implement Restore for ColumnDef
2. implement Restore for ColumnOption
(todo Waiting for FuncCallExpr)

---

Issue: [https://github.com/pingcap/tidb/issues/8532](https://github.com/pingcap/tidb/issues/8532)

### CODE REVIEW

File: ast/ddl.go [hunk 1]
```diff
@@ -369,7 +369,64 @@ type ColumnOption struct {
...
+			if err := n.Expr.Restore(ctx); err != nil {
+				return errors.Annotate(err, "An error occurred while splicing ColumnOption DefaultValue Expr")
+			}
+		}
+	case ColumnOptionUniqKey:
+		ctx.WriteKeyWord("UNIQUE KEY")
+	case ColumnOptionNull:
+		ctx.WriteKeyWord("NULL")
+	case ColumnOptionOnUpdate:
+		//todo Waiting for FuncCallExpr
```
Comment (leoppro) [inline ast/ddl.go]:
```suggestion
 		//TODO: Waiting for FuncCallExpr
```

File: ast/ddl.go [hunk 2]
```diff
@@ -369,7 +369,64 @@ type ColumnOption struct {
...
+		//ctx.WriteKeyWord("ON UPDATE")
+		//if n.Expr != nil {
+		//	ctx.WritePlain(" ")
+		//	if err := n.Expr.Restore(ctx); err != nil {
+		//		return errors.Annotate(err, "An error occurred while splicing ColumnOption ON UPDATE Expr")
+		//	}
+		//}
+	case ColumnOptionFulltext: //There is no parsing in goyacc
```
Comment (leoppro) [inline ast/ddl.go]:
```suggestion
 	case ColumnOptionFulltext: 
 	// TiDB Parser ignore the `ColumnOptionFulltext` type now
```
and i think we should throw a error here.

File: ast/ddl.go [hunk 3]
```diff
@@ -369,7 +369,64 @@ type ColumnOption struct {
...
+			}
+			ctx.WritePlain(")")
+		}
+	case ColumnOptionReference:
+		if n.Refer != nil {
+			if err := n.Refer.Restore(ctx); err != nil {
+				return errors.Annotate(err, "An error occurred while splicing ColumnOption ReferenceDef")
+			}
+		}
+	}
```
Comment (leoppro) [inline ast/ddl.go]: add default case and throw a error

File: ast/ddl_test.go [hunk 4]
```diff
@@ -164,6 +164,48 @@ func (ts *testDDLSuite) TestDDLConstraintRestore(c *C) {
...
+	}
+	extractNodeFunc := func(node Node) Node {
+		return node.(*CreateTableStmt).Cols[0].Options[0]
+	}
+	RunNodeRestoreTest(c, testCases, "CREATE TABLE child (id INT %s)", extractNodeFunc)
+}
+
+func (ts *testDDLSuite) TestDDLColumnDefRestore(c *C) {
+	testCases := []NodeRestoreTestCase{
+		{"id int(11) PRIMARY KEY", "`id` INT(11) PRIMARY KEY"},
```
Comment (leoppro) [inline ast/ddl_test.go]: please add  tests for `id int`,`id int null`, and other data types

File: ast/ddl_test.go [hunk 5]
```diff
@@ -164,6 +164,48 @@ func (ts *testDDLSuite) TestDDLConstraintRestore(c *C) {
...
 
+func (ts *testDDLSuite) TestDDLColumnOptionRestore(c *C) {
+	testCases := []NodeRestoreTestCase{
+		{"primary key", "PRIMARY KEY"},
+		{"not null", "NOT NULL"},
+		{"null", "NULL"},
+		{"auto_increment", "AUTO_INCREMENT"},
+		{"DEFAULT 10", "DEFAULT 10"},
+		{"DEFAULT '10'", "DEFAULT '10'"},
+		{"DEFAULT 1.1", "DEFAULT 1.1"},
```
Comment (leoppro) [inline ast/ddl_test.go]: test more default expr

File: ast/ddl.go [hunk 6]
```diff
@@ -369,7 +369,64 @@ type ColumnOption struct {
...
+		return nil
+	case ColumnOptionPrimaryKey:
+		ctx.WriteKeyWord("PRIMARY KEY")
+	case ColumnOptionNotNull:
+		ctx.WriteKeyWord("NOT NULL")
+	case ColumnOptionAutoIncrement:
+		ctx.WriteKeyWord("AUTO_INCREMENT")
+	case ColumnOptionDefaultValue:
+		ctx.WriteKeyWord("DEFAULT")
+		if n.Expr != nil {
```
Comment (leoppro) [inline ast/ddl.go]: I don't think we need judge the `Expr` isn't `nil`,
